build/mech_converter.py

```diff
@@ -170,257 +218,287 @@ def convert(input_file, mech_dir, mcm_dir):
 
-
-    # Convert peroxy_radicals to a list of strings, each of the RO2 species from 'Peroxy radicals'
+    # Convert peroxy_radicals to a list of strings, with each of the RO2 species from the RO2 sum
+    # in the 'Peroxy radicals' section.
     ro2List = []
     for item in peroxy_radicals:
-        if not re.match('\*', item):
-            # We have an equals sign on the first line. Handle this by splitting against =, then taking the last element of the
-            # resulting list, which will either be the right-hand side of the first line, or the whole of any other line.
-            # Similarly, the final line will end with a colon. Handle in a similar way.
-            # Then split by +. Append each item to ro2_input: multiple appends use 'extend'
-            ro2List.extend([elem.strip() for elem in item.split('=')[-1].split(';')[0].strip().split('+')])
-    # Remove empty strings
+        if not re.match(r'\*', item):
+            # We have an equals sign on the first line. Handle this by splitting against '=',
+            # then taking the last element of the resulting list, which will either be the
+            # right-hand side of the first line, or the whole of any other line. Similarly,
+            # the final line will end with a colon. Handle in a similar way. Then split
+            # by '+'. Append each item to ro2_input: multiple appends use 'extend'.
+            ro2List.extend([elem.strip() for elem in \
+                            item.split('=')[-1].split(';')[0].strip().split('+')])
+    # Remove empty strings.
     ro2List = list(filter(None, ro2List))
 
-
-    # Read in the reference RO2 species from the peroxy-radicals_v3.3.1 file
-    with open(os.path.join(mcm_dir, 'peroxy-radicals_v3.3.1'), 'r') as RO2List_file:
+    # Read in the reference list of RO2 species from the MCM (peroxy-radicals_v*).
+    #
+    # => the RO2 reference list is specific to each version of the MCM
+    # => RO2 lists for versions v3.1, v3.2, v3.3.1 of the MCM are available
+    # => change the filename if using a version of the MCM other than the default (v3.3.1)
+    #
+    # TODO: implement a different way to set the mcm version (see issue #297)
+    with open(os.path.join(mcm_vers, 'peroxy-radicals_v3.3.1'), 'r') as RO2List_file:
         RO2List_reference = [r.rstrip() for r in RO2List_file.readlines()]
 
-    # Check each of the RO2s from 'Peroxy radicals' are in the reference RO2 list. If not print a warning at the top of
-    # mechanism.f90 for each errant species.
-    # TODO: This will break the exected format when mechanism.f90 is replaced by a parsable format.
+    # Check that each of the RO2s from 'Peroxy radicals' are present
+    # in the RO2 reference list from the MCM. If not, print a warning
+    # at the top of mechanism.f90 for each errant species.
+    #
+    # TODO: This will break the expected format when mechanism.f90 is
+    #       replaced by a parsable format
```

Comment:
This TODO comment was added with PR #347. I think it refers to the implementation of the function parser which then was abandoned in favor of the shared `.so` file (?)
If so, it's probably not something "to do" anymore?

Reply:
I think so - fine to delete.